plugin/src/main/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfigurator.java

```diff
@@ -145,10 +145,16 @@ private T tryConstructor(Constructor<T> constructor, Mapping config, Configurati
             for (Object arg : args) {
                 argumentTypes.add(arg != null ? arg.getClass().getName() : "null");
             }
+            List<String> parameterTypes = new ArrayList<>(parameters.length);
+            for (Parameter parameter: parameters){
+                parameterTypes.add(parameter.getParameterizedType().getTypeName());
+            }
             throw new ConfiguratorException(this,
                     "Failed to construct instance of " + target +
                             ".\n Constructor: " + constructor.toString() +
-                            ".\n Arguments: " + argumentTypes, ex);
+                            ".\n Arguments: " + argumentTypes +
+                            ".\n Expected ParamNames: " + names +
```

Comment:
Can you add a test which verifies this please?
Looks correct but without seeing the output it's hard to tell.

Also can you show a sample exception here?

Reply:
One minute just let me try adding a test

Reply:
@timja Now that @casz has added the test 

> @sladyn98 we are thinking it right here:
> 
> https://github.com/jenkinsci/configuration-as-code-plugin/blob/73059f9d1557220787b75328cafe79849f5ff5d7/plugin/src/test/java/io/jenkins/plugins/casc/impl/configurators/DataBoundConfiguratorTest.java#L103-L109

Because we would not be able to verify the params in this case @casz